src/Pages/ReferenceTokens.razor

```diff
@@ -44,11 +47,96 @@
             </div>
         </div>
     </TabPane>
+    <TabPane TabId="check" TabIndex="1">
+        <div class="row">
+            <div class="col-6">
+                <EditForm Model="_checkTokenModel" OnSubmit="@HandleSubmit" autocomplete="off">
+                    <div class="from-group mb-3">
+                        <label for="clientId">Client Id</label>
+                        <select id="clientId" class="form-select" @bind="_checkTokenModel.ClientId" required>
+                            <option value="" disabled selected>Select Client</option>
+                            @foreach (var client in ConfigurationDbContext.Clients)
+                            {
+                                <option value="@client.ClientId">@client.ClientId</option>
+                            }
+                        </select>
+                    </div>
+                    <div class="form-group mb-3">
+                        <label for="referenceToken">Reference Token</label>
+                        <input type="text" id="referenceToken" class="form-control" autocomplete="false" @bind="_checkTokenModel.ReferenceToken" required/>
+                    </div>
+                    <button type="submit" class="btn btn-sm btn-primary" role="button">Check</button>
+                </EditForm>
+            </div>
+            <div class="col-3">
+                @switch (_referenceTokenData.Status)
+                {
+                    case "notChecked":
+                        break;
+                    case "Invalid":
+                        <div class="alert alert-danger" role="alert">
+                            <h4 class="alert-heading">Invalid</h4>
+                            <p>The Reference Token for @_referenceTokenData.ClientId is invalid</p>
+                        </div>
+                        break;
+                    case "Active":
+                        <div class="alert alert-success" role="alert">
```

Comment:
The content for each case is almost the same here. Perhaps we could wrap this into a separate component (ie. `ReferenceTokenStatus`) which we can pass an instance of `ReferenceTokenDataModel` that will then render itself as a `<div class="alert...`.

Reply:
changed in 6dfd93d